docs/design/coreclr/jit/jit-disasm.md

```diff
@@ -0,0 +1,71 @@
+* `DOTNET_JitDisasmDiffable` - set to 1 to make the generated code "diff-able", namely, replace pointer values in the output with
+the same well-known, identical values, so they textually compare identically.
+* `DOTNET_JitDisasmWithAlignmentBoundaries` - set to 1 to display alignment boundaries in the generated code.
+* `DOTNET_JitDisasmWithCodeBytes` - set to 1 to display the actual code bytes in addition to textual disassembly.
+* `DOTNET_JitDisasmWithGC` - Debug/Checked builds only. Set to 1 to display GC information interleaved with the textual disassembly.
+* `DOTNET_JitDisasmWithDebugInfo` - Debug/Checked builds only. Set to 1 to display debug information (variable live ranges)
+interleaved with the textual disassembly.
+* `DOTNET_JitStdOutFile` - if not set, all JIT output goes to standard output. If set, it is the name of a file to which JIT output
+will be written.
+
+# The JIT late disassembler
+
+The disassembly displayed by `DOTNET_JitDisasm` is printed based on a JIT internal representation. An additional disassembler, called
+the "late disassembler", is available that disassembles the final code bytes. The late disassembler uses the coredistools package
+(based on LLVM) to interpret the code bytes. In this way, it provides a way to verify that the JIT disassembly matches a third-party
+"oracle" to agree on the disassembly for a particular set of code bytes.
+
+The late disassembler currently is only available in Debug/Checked builds.
+
+To invoke the late disassembler, use:
+* `DOTNET_JitLateDisasm` - output late disassembly for the specified functions. E.g., `DOTNET_JitLateDisasm=Main`,
+`DOTNET_JitLateDisasm=Main Test1 Test2`, `DOTNET_JitLateDisasm=*`.
+* `DOTNET_JitLateDisasmTo` - the late disassembler will write output to standard output, or if `DOTNET_JitStdOutFile` is set, to
+that file. However, if `DOTNET_JitLateDisasmTo` is set, it will be used as the name of a file where late disassembly output will be sent.
+
+The late disassembler will load coredistools (currently, version 1.4.0) and have it disassemble all the generated code bytes.
+
+## Using the late disassembler with the emitter unit tests
+
+One use for the late disassembler is when adding new instructions to the JIT emitter, assuming coredistools already knows about those
+new instructions. For example, when adding Arm64 SVE instructions, new unit tests for each new instruction are added to
+`CodeGen::genEmitterUnitTests()` (or a function called by that). Then, create a "Hello World" program with a `Main` function and set:
+
+* `DOTNET_JitDisasm=Main`
+* `DOTNET_JitLateDisasm=Main`
+* `DOTNET_JitLateDisasmTo=latedis.asm` - optionally send `JitLateDisasm` output to a separate file.
+* `DOTNET_JitEmitUnitTests=Main`
+* `DOTNET_JitEmitUnitTestsSections=sve`
+
+With these environment variables set, run the program. The `Main` function will be populated with a lot of additional
```

Comment:
If we specify `DOTNET_JitStdOutFile`, then does it overrides `DOTNET_JitLateDisasmTo` file? I was wondering how we can produce 2 files for above scenario, the one that contains `JITDisasm` output and other that has `latedis.asm`. Currently, the way it is written, seems like we have `latedis.asm` get populated and redirect the `JitDisasm` output using 

```cmd
corerun foo.dll > jitdisasm.asm
```